### ISSUE TITLE
unsigned expression < 0 is always false

### ISSUE THREAD

**Issue by chriscamacho** [ooxi/xml.c#24]
src/xml.c: In function 'xml_parser_error':
src/xml.c:318:24: error: comparison of unsigned expression < 0 is always false [-Werror=type-limits]
  318 |  #define max(X,Y) ((X) > (Y) ? (X) : (Y))
      |                        ^
src/xml.c:319:21: note: in expansion of macro 'max'
  319 |  size_t character = max(0, min(parser->length, parser->position + offset));
      |

**Comment by ooxi** [comment]
You are right, those calculations most likely have to be done using `ssize_t` instead of `size_t`. Would you like to propose a patch?

**Comment by chriscamacho** [comment]
in my own project I just hacked the line to....
`size_t character = min(parser->length, parser->position + offset);`
but for all I know that might make things explode....
